### PR TITLE
Register watchers

### LINKED ISSUE #4416: OGIP 36: Antragssteller und Gremienverantwortlichen als Watcher registrieren
Antragssteller und Gremienverantwortliche sollen als Watcher der Anträge und eingereichter Anträge  registriert werden.

- Antragssteller sollen bei Erstellung und Modifikation eines Antrags im Dossier als Watcher hinzugefügt werden. Ändert sich der Antragssteller beim Bearbeiten muss dies ebenfalls aktualisiert werden, sprich der alte Watcher muss entfernt werden und der neue Antragssteller soll statdessen als Watcher eingetragen werden.
- Gremienverantwortliche sollen beim Einreichen eines Antrags als Watcher des eingereichten Antrags in der Rolle Gremien-Verantwortlicher registriert werden. Gremienverantworkliche werden auf dem Gremium als LDAP-Gruppe konfiguriert, `Committee.group_id`. Alle Benutzer dieser Gruppe sollen beim **Erstellen** des eingereichten Antrags als Watcher registriert werden. (Dies bedeutet, dass Änderungen an der Gruppenmitgliedschaft erst für neue Anträge gelten. Wir setzen dies bewusst so um)
- Migration Anträge: Alle Antragssteller aller bestehenden Anträge als Watcher in der Rolle Antragssteller registrieren (Kann unabhängig von https://github.com/4teamwork/opengever.core/issues/4412 umgesetzt werden, da auch der `Creator` verwendet werden kann.
- Migration Eingereichter Anträge: Alle Benutzer der Gruppe der Gremien-Verantwortlichen sollen als Watcher in der Rolle Gremien-Verantwortlicher registriert werden.

### PR DESCRIPTION
**Pull request by elioschmutz**
Handle watchers for proposals.

⚠️ `get_users_by_group` is very slow. Is there another way, how to get all users of a group? Is it cached somewhere?

closes #4416

### CODE REVIEW

**Comment by elioschmutz** [comment]
@deiferni

File: opengever/core/upgrades/20180621190112_register_watchers_for_proposals/upgrade.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class RegisterWatchersForProposals(SchemaMigration):
+    """Register watchers for proposals.
+    """
+
+    def migrate(self):
+        for obj in self.objects(
+            {'portal_type': ['opengever.meeting.proposal', 'opengever.meeting.submittedproposal']},
+                'Register watchers for proposals'):
+
+            if ISubmittedProposal.providedBy(obj):
```
Comment (deiferni) [inline opengever/core/upgrades/20180621190112_register_watchers_for_proposals/upgrade.py]: i'd rather do two separate queries for both types

File: opengever/meeting/activity/watchers.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+from opengever.activity import notification_center
+from opengever.activity.roles import COMMITTEE_RESPONSIBLE_ROLE
+from opengever.activity.roles import PROPOSAL_ISSUER_ROLE
+from opengever.meeting.activity.helpers import get_users_by_group
+
+
+def add_watchers_on_submitted_proposal_created(obj):
```
Comment (deiferni) [inline opengever/meeting/activity/watchers.py]: hmm i think you should implement the activities as classes that interit from `BaseActivity` (https://github.com/4teamwork/opengever.core/blob/4cc632fefd44a368a64085a367c179f8cc013cbc/opengever/activity/base.py#L6) see https://github.com/4teamwork/opengever.core/blob/4cc632fefd44a368a64085a367c179f8cc013cbc/opengever/task/activities.py for an example.

File: opengever/meeting/activity/watchers.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from opengever.activity import notification_center
```
Comment (deiferni) [inline opengever/meeting/activity/watchers.py]: there seems to be the convention of putting activity related things into an `activitites.py` module. could you move it?

File: opengever/meeting/activity/helpers.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from plone import api
+from plone.api.exc import GroupNotFoundError
+
+
+def get_users_by_group(groupname):
```
Comment (deiferni) [inline opengever/meeting/activity/helpers.py]: as this is non-security related information i would use `ogds_service` to retrieve this information, which will be much faster.

**Comment by elioschmutz** [comment]
@deiferni This is not the activity-implementation. This PR only registers the watchers.

The activities will be implemented in the next PR: https://github.com/4teamwork/opengever.core/pull/4527

So I think, there is no need to move or change the current implementation.

File: opengever/meeting/activity/helpers.py (hunk 4 above)
Comment (deiferni) [inline opengever/meeting/activity/helpers.py]: this should use the ogds though

**Comment by elioschmutz** [comment]
@deiferni 👍 ready to merge

File: opengever/meeting/activity/helpers.py (hunk 4 above)
Comment (elioschmutz) [reply opengever/meeting/activity/helpers.py]: 👍 updated

**Pull request merged**